os/drivers/lcd/st7701sn.c

```diff
@@ -18,8 +18,19 @@
 
 #include <tinyara/lcd/st7701sn.h>
 #include <mipi_lcd.h>
+#include <tinyara/lcd/lcd_dev.h>
+#include <math.h>
 
 int check_lcd_vendor_send_init_cmd(struct mipi_lcd_dev_s *priv)
 {
 	return send_init_cmd(priv, lcd_init_cmd_g);
 }
+
+int get_lcdinfo(FAR struct lcd_info_s *lcdinfo)
+{
+	lcdinfo->lcd_height_mm = (uint8_t)(93.6 + 0.5f);
+	lcdinfo->lcd_width_mm = (uint8_t)(56.16 + 0.5f);
+	lcdinfo->lcd_size_inch = sqrtf(93.6 * 93.6 + 56.16 * 56.16) / 25.4f;
+	lcdinfo->lcd_dpi = sqrtf(CONFIG_LCD_XRES * CONFIG_LCD_XRES + CONFIG_LCD_YRES * CONFIG_LCD_YRES) / lcdinfo->lcd_size_inch;
```

Comment:
Please use definition 
```c
#define LCD_HEIGHT_MM   93.6
#define LCD_WIDTH_MM    56.16
```

and calibrating lcd_size_inch, lcd_dpi is common logic, please move to lcd_dev